README.md

```diff
@@ -2,31 +2,318 @@
+                return employee;
+            }
+
+        }
+        catch (SQLException e) { System.out.println(e.getMessage());}
+
+        //The employee was not in the table
+        return null;
+    }
+
+
+    public static void main(String[] args)  {
+        try {
+            Connection connection  = DriverManager.getConnection(DB_URL, USER, PASSWORD);
+            Employee employee;
+
+            //Get employee with id=100
+            employee = find(connection, 100);
+            System.out.println(employee);  //Employee{id=100, email='emp100@company.com', office='a456', salary=99000.0}
+
+            //Get non-existent employee id=99
+            employee = find(connection, 99);
+            System.out.println(employee); //null
+
+            connection.close();
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+}
+```
+
+Run `FindEmployee.main` to confirm the output, which shows the SELECT statement that is executed
+for each call to the `find` method:
+
+```text
+SELECT * FROM employee WHERE id = 100
+Employee{id=100, email='emp100@company.com', office='a456', salary=99000.0}
+SELECT * FROM employee WHERE id = 99
+null
+```
+
+
+## Saving an array of `Employee` objects
+
+
+
+The JDBC `PreparedStatement` interface is a special kind of  `Statement` that improves performance
+when an SQL statement needs to be executed multiple times:
+
+- The SQL statement may include value placeholders or parameters using the `?` character.
+- The SQL statement may be reused with new parameter values specified for each `?`.
+
+The `PersistEmployeeArray` class below contains a `persist` method that takes an `Employee` array
+as a parameter, and uses the `PreparedStatement` interface to add each employee to the table
+using the same statement object.
+
+- The  method creates an instance of `PreparedStatement` that contains
+  placeholders `?` for each column in the `employee` table.
+- The loop assigns the placeholders to the instance variable values of the next `Employee` object in the array.
+- The prepared statement object executes the insert statement to insert the employee row. 
+
+
+```java
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class PersistEmployeeArray {
+
+    static final String DB_URL = "jdbc:postgresql://localhost:5432/employee_db";
+    static final String USER = "postgres";
+    static final String PASSWORD = "postgres";
+
+    public static void persist(Connection connection, Employee[] employees) throws SQLException {
+        //Create an insert statement with placeholder ? for each column value
+        String insertStatement = "INSERT INTO employee (id, email, office, salary) VALUES (?,?,?,?)";
+
+        //Create a PersistEmployeeArray object for sending parameterized SQL statements through the database connection.
+        java.sql.PersistEmployeeArray preparedStmt = connection.prepareStatement(insertStatement);
```

Comment:
Is the left side of this line supposed to be `java.sql.PersistEmployeeArray`? Or should it be `PreparedStatement`?

Reply:
Yes!  Thanks for catching that error!

Reply:
Should be PreparedStatement